### PR TITLE
ARC-1514-audience-tech-planning-create-1-x-donor-counts-actuals-and-budget-mart

### PR DESCRIPTION
**Pull request by ryantimjohn**
# Pull Request for dbt-arc-functions

- [x] Open the pull request as a draft so that you can run tests against the code and inspect the files in the Files view of the Pull Request interface in Github.

- [x] Once you're satisfied, convert the draft into an open Pull Request by clicking "Ready for Review" near the bottom of this page. You should not make any changes to the code except in dire emergency once you've converted from draft, so make sure you're really satisfied.

- [x] Add two of the principal contributors to this PR for review. Currently, the principal contributors are @dmbluestate , @Frydafly , and @ryantimjohn.

- [x] We've divided our PR templates into Macros, Utils and Docs PR requests. Delete the ones you're not using.
### Macros (delete if not using)

- [x] Link Jira tickets (or Github Issues) to PR here. Make sure that the Jira ticket has a good description as to why this PR is necessary and the business problem that this PR is solving. Moreover, if there have been any discussions of this ticket in Slack, make sure they're linked in the Jira ticket:

https://bluestate.atlassian.net/browse/ARC-1514

- [x] Make sure this is named using Jira extension so that naming convention is consistent. If it's not, that's okay. Go to the Jira ticket, click into the ticket, then on the right under Details, find Development, then click the Create branch button. Copy the Branch Name from there. Read here [how to rename a branch in Github](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/managing-branches-in-your-repository/renaming-a-branch) to rename this branch. In the future, create branches from Jira.

- [x] Make sure the Macro is documented by running the notebook: `create_docs.ipynb` util in this repo. From there, you'll have to either fill in descriptions by hand OR use our `fill_in_descriptions_using_openai.ipynb` to fill in descriptions using OpenAI's ChatGPT. Always make sure to check over ChatGPT's column descriptions, as they're likely not perfect.

- [x] Test branch on a development branch of an existing client (ideally the one that raised the bug). Do this by changing revision of the package to the branch name in `packages.yml` file in dbt State client below.

https://github.com/bsd/arc-uusa-dbt/pull/32

- [x] If new macro files were created or updated, re-run `create_or_update_standard_models.py` for the client dbt project, replacing models and dependencies

- [x] check in dbt that client package successfully runs `dbt deps`, `dbt compile`, and `dbt run`; screenshot this page

Screenshot 2023-08-31 at 3 05 04 PM

- [x] query the resulting staging model in bigquery (or by previewing mart in dbt cloud, or your IDE); screenshot this

Screenshot 2023-09-20 at 9 23 20 AM

Screenshot 2023-09-20 at 9 20 42 AM

### CODE REVIEW

**Comment by ryantimjohn** [comment]
@Frydafly it's ready for you now, you can do some exploratory data analysis. Here's table query:
```sql
SELECT * 
FROM `bsd-arc-uusa.dev_marts.mart_audience_budget_with_audience_transaction` 
LIMIT 1000
```

**Comment by ryantimjohn** [comment]
Note on this query:
```sql
SELECT * 
FROM `bsd-arc-uusa.dev_marts.mart_audience_budget_with_audience_transaction` 
WHERE 
total_onetime_donor_counts > 0
AND onetime_donor_count_budget > 0
```
that only 86 rows come up where counts is not null and budget is > 0, I don't know if this is expected behavior or not @Frydafly

File: macros/audience_budget_with_audience_transaction/marts/mart_audience_budget_with_audience_transaction.sql
Comment (dmbluestate) [inline macros/audience_budget_with_audience_transaction/marts/mart_audience_budget_with_audience_transaction.sql:1]: May improve readability by adding the dimensions above the metrics in this query.

**Review by dmbluestate — APPROVED** [review]

File: macros/audience_budget_with_audience_transaction/marts/mart_audience_budget_with_audience_transaction.sql
Comment (Frydafly) [reply macros/audience_budget_with_audience_transaction/marts/mart_audience_budget_with_audience_transaction.sql:1]: strongly agree

File: macros/audience_budget_with_audience_transaction/marts/mart_audience_budget_with_audience_transaction.sql [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        coalesce(onetime_donor_counts.platform, audience_budget.join_source) as platform
+    from {{ ref(reference_0_name) }} as onetime_donor_counts
+    full join
+        {{ ref(reference_1_name) }} as audience_budget
+        on onetime_donor_counts.date_day = audience_budget.date_day
+        and upper(onetime_donor_counts.interval_type)
+        = upper(audience_budget.interval_type)
+        and upper(onetime_donor_counts.donor_audience)
+        = upper(audience_budget.donor_audience)
+        and upper(onetime_donor_counts.platform) = upper(audience_budget.join_source)
```
Comment (Frydafly) [inline macros/audience_budget_with_audience_transaction/marts/mart_audience_budget_with_audience_transaction.sql:50]: @dmbluestate

**Review by Frydafly — APPROVED** [review]
LGTM -- pending confirmation platform/channel is matching on the actuals and budget side

File: macros/audience_budget_with_audience_transaction/marts/mart_audience_budget_with_audience_transaction.sql (hunk 1 above)
Comment (ryantimjohn) [reply macros/audience_budget_with_audience_transaction/marts/mart_audience_budget_with_audience_transaction.sql:50]: Ack, this was an auto format from dbt, will keep my eye out for these

**Pull request merged**